reconcile/quay_membership.py

```diff
@@ -1,51 +1,61 @@
 import logging
 import sys
-
-from reconcile.utils import gql
-from reconcile.utils import expiration
-
+from typing import Optional, Sequence, Union
+
+from reconcile.gql_queries.quay_membership.quay_membership import (
+    BotV1,
+    PermissionQuayOrgTeamV1,
+    QuayMembershipQueryData,
+    RoleV1,
+    UserV1,
+    query_string,
+)
+from reconcile.quay_base import get_quay_api_store
+from reconcile.status import ExitCodes
+from reconcile.utils import expiration, gql
 from reconcile.utils.aggregated_list import (
-    AggregatedList,
     AggregatedDiffRunner,
+    AggregatedList,
     RunnerException,
 )
-from reconcile.quay_base import get_quay_api_store
-from reconcile.status import ExitCodes
 from reconcile.utils.quay_api import QuayTeamNotFoundException
 
-QUAY_ORG_QUERY = """
-{
-  roles: roles_v1 {
-    name
-    users {
-      name
-      quay_username
-    }
-    bots {
-      name
-      quay_username
-    }
-    permissions {
-      service
-      ...on PermissionQuayOrgTeam_v1 {
-        quayOrg {
-          name
-          instance {
-            name
-          }
-        }
-        team
-      }
-    }
-    expirationDate
-  }
-}
-"""
-
 QONTRACT_INTEGRATION = "quay-membership"
 
 
-def process_permission(permission):
+def get_permissions_for_quay_membership() -> list[PermissionQuayOrgTeamV1]:
+    gqlapi = gql.get_api()
+    query_data = gqlapi.query(query_string())
+    quay_membership = QuayMembershipQueryData(**query_data)
+    if not quay_membership.permissions:
+        return []
+    return [
+        p for p in quay_membership.permissions if isinstance(p, PermissionQuayOrgTeamV1)
+    ]
+
+
+def expiration_filter(
+    roles: Optional[list[RoleV1]], key: str = "expiration_date"
+) -> list[RoleV1]:
+    """Filters roles and returns the ones which are not yet expired.
+
+    adaption of reconcile.utils.expiration.filter which isn't GQL classes ready yet.
```

Comment:
i would extract this to a separate PR. there is no urgency to this work, hence we shouldn't accumulate technical debt. we can take the time and put this function where it belongs for future use.

Reply:
done